### PR TITLE
Some optimizations

### PR DESCRIPTION
**Pull request by Taiyou06**
- Improve Mth.clamp performance by 1.2 - 1.4x
- Use MCUtil.asyncExecutor for MAIN_WORKER_EXECUTOR
- Lithium IterateOutwardsCache
- Lithium HashedList for TileEntities
- Lithium CompactSineLUT
- Lithium fast_util lookup table for Direction.Axis
- C2ME Reduce Allocations on OreFeature
- Lithium fast elytra & hand check

Sorry if the patch descriptions are lacking, 1st time PRing to a fork and working with patch system.

### CODE REVIEW

File: patches/server/0119-Multithreaded-Tracker.patch
Comment (HaHaWTH) [inline patches/server/0119-Multithreaded-Tracker.patch:1]: This commit reverts the fix, please remove thread instance check

File: patches/server/0126-Asynchronous-locator.patch
Comment (HaHaWTH) [inline patches/server/0126-Asynchronous-locator.patch:1]: Same as patch 0119( ´▽` )

File: patches/server/0136-C2ME-Reduce-Allocations.patch
Comment (HaHaWTH) [inline patches/server/0136-C2ME-Reduce-Allocations.patch:1]: getCachedOrNewBits class should be renamed to GetCachedOrNewBits or CachedOrNewBitsGetter

File: patches/server/0138-Lithium-CompactSineLUT.patch
Comment (HaHaWTH) [inline patches/server/0138-Lithium-CompactSineLUT.patch:1]: CompactSineLUT is slower in Dreeam's test before, need benchmark this again

File: patches/server/0139-Lithium-IterateOutwardsCache.patch
Comment (HaHaWTH) [inline patches/server/0139-Lithium-IterateOutwardsCache.patch:1]: Do not use import here, use full-qualified name plz♪(^∇^*)

File: patches/server/0141-Use-MCUtil.asyncExecutor-for-MAIN_WORKER_EXECUTOR.patch
Comment (HaHaWTH) [inline patches/server/0141-Use-MCUtil.asyncExecutor-for-MAIN_WORKER_EXECUTOR.patch:1]: Same as patch 0139

**Review by HaHaWTH — APPROVED** [review]

**Comment by Dreeam-qwq** [comment]
Good job guys!

**Pull request merged**